client/sites/components/sites-dashboard.tsx

```diff
@@ -193,41 +192,58 @@ const SitesDashboard = ( {
 				},
 			},
 		},
-	} as DataViewsState;
-	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( defaultDataViewsState );
+	} as View;
+	const [ viewState, setViewState ] = useState< View >( defaultViewState );
+	const [ selectedItem, setSelectedItem ] = useState< SiteExcerptData | null >( null );
 
 	useEffect( () => {
 		const fields = getFieldsByBreakpoint( isDesktop );
 		const fieldsForBreakpoint = [ ...fields ].sort().toString();
-		const existingFields = [ ...( dataViewsState?.fields ?? [] ) ].sort().toString();
+		const existingFields = [ ...( viewState?.fields ?? [] ) ].sort().toString();
 		// Compare the content of the arrays, not its referrences that will always be different.
 		// sort() sorts the array in place, so we need to clone them first.
 		if ( existingFields !== fieldsForBreakpoint ) {
-			setDataViewsState( ( prevState ) => ( { ...prevState, fields } ) );
+			setViewState( ( prevState ) => ( { ...prevState, fields } ) );
 		}
 
 		const siteNameColumnWidth = getSiteNameColWidth( isDesktop, isWide );
-		if ( dataViewsState.layout.styles.site.width !== siteNameColumnWidth ) {
-			setDataViewsState( ( prevState ) => ( {
+
+		if (
+			viewState.type === 'table' &&
+			viewState.layout?.styles?.site?.width !== siteNameColumnWidth
+		) {
+			// @ts-expect-error -- Need to fix the layout type upstream in @wordpress/dataviews to support styles.
+			setViewState( ( prevState ) => ( {
 				...prevState,
 				layout: {
 					styles: {
-						...prevState.layout.styles,
+						// @ts-expect-error -- Need to fix the layout type upstream in @wordpress/dataviews to support styles.
+						...prevState.layout?.styles,
 						site: {
 							width: siteNameColumnWidth,
 						},
 					},
 				},
 			} ) );
 		}
-	}, [ isDesktop, isWide, dataViewsState?.fields, dataViewsState?.layout?.styles?.site?.width ] );
+	}, [
+		isDesktop,
+		isWide,
+		viewState.type,
+		viewState?.fields,
+		// @ts-expect-error -- Need to fix the layout type upstream in @wordpress/dataviews to support styles.
+		viewState?.layout?.styles?.site?.width,
+	] );
+
+	useSyncSelectedSite( viewState, setViewState, selectedSite );
 
-	useSyncSelectedSite( dataViewsState, setDataViewsState, selectedSite );
+	// todo: add this
+	// const siteId = useSelector( getSelectedSiteId );
```

Comment:
Should probably still do this, but might not need it anymore? Need to figure out why site selection is no longer updating the url or if loading the /overview/:site url directly doesn't load flyout content.

Reply:
Worked this concept in elsewhere.